### PR TITLE
Fix search error documentation via error code

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

Currently, we cannot search the errors documentation via error code prefix, this PR fixes this issue.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

```
$ tiup errdoc BR:Restore
# Error: BR:Restore:ErrRestoreChecksumMismatch

  restore checksum mismatch


# Error: BR:Restore:ErrRestoreInvalidBackup

  invalid backup


# Error: BR:Restore:ErrRestoreInvalidRange

  invalid restore range


# Error: BR:Restore:ErrRestoreInvalidRewrite

  invalid rewrite rule


# Error: BR:Restore:ErrRestoreModeMismatch

  restore mode mismatch


# Error: BR:Restore:ErrRestoreNoPeer

  region does not have peer


# Error: BR:Restore:ErrRestoreRangeMismatch

  restore range mismatch


# Error: BR:Restore:ErrRestoreRejectStore

  failed to restore remove rejected store


# Error: BR:Restore:ErrRestoreResolvedTsConstrain

  resolved ts constrain violation


# Error: BR:Restore:ErrRestoreSchemaNotExists

  schema not exists


# Error: BR:Restore:ErrRestoreSplitFailed

  fail to split region


# Error: BR:Restore:ErrRestoreTableIDMismatch

  restore table ID mismatch


# Error: BR:Restore:ErrRestoreWriteAndIngest

  failed to write and ingest


13 matched

```

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: components/errdoc/main.go [hunk 1]
```diff
@@ -58,19 +58,53 @@ func searchError(args []string) error {
...
+	}
+	var terms []query.Query
+	var prefix []query.Query
+	for _, arg := range args {
+		terms = append(terms, bleve.NewTermQuery(arg))
+		prefix = append(prefix, bleve.NewPrefixQuery(arg))
+	}
+	queries = append(queries, bleve.NewConjunctionQuery(terms...))
+	queries = append(queries, bleve.NewConjunctionQuery(prefix...))
+
```
Comment (lucklove) [inline components/errdoc/main.go:73]:
```golang
var qs []query.Query
for _, arg := range args {
     qs = append(qs, bleve.NewTermQuery(arg), bleve.NewPrefixQuery(arg))
}
queries = append(queries, bleve.NewConjunctionQuery(qs...))
```

File: components/errdoc/main.go [hunk 2]
```diff
@@ -58,19 +58,53 @@ func searchError(args []string) error {
...
+				all[code] = struct{}{}
+			}
 		}
 	}
 
+	var sorted []string
+	for code := range all {
+		sorted = append(sorted, code)
+	}
+	sort.Slice(sorted, func(i, j int) bool {
```
[the remark this answers is not in the document]
Comment (lonng) [reply components/errdoc/main.go:98]: I think the query result should be stable.

File: components/errdoc/main.go (hunk 1 above)
Comment (lonng) [reply components/errdoc/main.go:73]: The `terms` and `prefix` are individual and apply to different scenarios. e.g:
1. terms: `commit transaction` can match the content `commit the transaction`, we use `commit` and `transaction` to match the content first and conjunction them.
2. prefix: `commit trans` can match the content `commit the transaction`, we use `commit` and `trans` to match the content first and conjunction them.

**Review by lucklove — APPROVED** [review]
LGTM

**Pull request merged**